File: Java Kata/VendingMachine.java
package com.audition;

import static com.audition.VendingMachine.COIN.DIME;
import static com.audition.VendingMachine.COIN.NICKEL;
import static com.audition.VendingMachine.COIN.QUARTER;

public class VendingMachine {
    public Inventory inventory;
    public double changeIn;



    public VendingMachine(){
        this.inventory = new Inventory();
        this.changeIn =  0.0;
    }

    public enum COIN {
        NICKEL(5.0, 0.835), DIME(2.268, 0.705), QUARTER(5.670, 0.955);

        private final double weight;
        private final double diameter;

        COIN(double weight, double diameter) {
            this.weight = weight;
            this.diameter = diameter;
        }

        boolean isEqual(double weight, double diameter) {
            if(weight == this.weight && diameter == this.diameter) {
                return true;
            }
            return false;
        }
    }

    public void insertMoney(Coin coin){
        try {
            double value = this.render(coin);
            this.increaseChangeInMachine(value);
        }
        catch(IllegalArgumentException e){
            System.out.println("Coin is invalid");
        }
    }

    public void increaseChangeInMachine(double change){
        this.changeIn = this.changeIn + change;

    }

    public double render(Coin coin) {
        double weight = coin.getWeight();
        double diameter = coin.getDiameter();
        if (NICKEL.isEqual(weight, diameter)) {
            return .05;
        } else if (DIME.isEqual(weight, diameter)) {
            return .10;
        } else if (QUARTER.isEqual(weight, diameter)) {
            return .25;
        } else {
            throw new IllegalArgumentException("Coin not recognized by system");
        }
    }

    public String buyProduct(char buttonPressed) {
        String outOfStock = "OUT OF STOCK";
        double price = 0;
        switch (buttonPressed) {
            case 'a':
                price = STOCK_VALUES.COLA.price;
                if(inventory.getColaStock() >0) {
                    if(this.changeIn >= price) {
                        this.subtractFromChangeIn(price);
                        this.inventory.substractColaStock();
                        return STOCK_VALUES.COLA.getName();
                    }
                    return "PLEASE INSERT MORE MONEY";
                }

                return outOfStock;

            case 'b':
                price = STOCK_VALUES.CHIP.price;
                if(inventory.getChipStock() > 0) {
                    if(this.changeIn >= price) {
                        this.subtractFromChangeIn(price);
                        this.inventory.subtractChipStock();
                        return STOCK_VALUES.CHIP.getName();
                    }
                    return "PLEASE INSERT MORE MONEY";
                }
                return outOfStock;

            case 'c':
                price = STOCK_VALUES.CANDY.price;
                if(inventory.getCandyStock() > 0) {
                    if(this.changeIn == price) {
                        this.subtractFromChangeIn(price);
                        this.inventory.subtractCandyStock();
                        return STOCK_VALUES.CANDY.getName();
                    }
                    else if(this.changeIn > price){
                        return "EXACT CHANGE ONLY PLEASE";
                    }
                    return "PLEASE INSERT MORE MONEY";
                }
                return outOfStock;
            default:
                return "ERROR";
        }
    }

    private void subtractFromChangeIn(double productAmount){
        this.changeIn = this.changeIn - productAmount;
    }

    public double getRemainingChange(){
        return this.changeIn;
    }
}

File: Javascript Kata/VendingMachine.js

var COIN = {'Weight': '', 'Diameter': ''};

const COINS = {'NICKEL': {'Weight': 5.000, 'Diameter': 0.835},
             'DIME' : {'Weight': 2.268, 'Diameter': 0.705},
             'QUARTER': {'Weight': 5.670 , 'Diameter': 0.955}};

class VendingMachine{

    constructor(){
        this.changeIn = 0.0;
        this.inventory = {"COLA": {'count': 10, 'price' : 1 },
                          "CHIP": {'count': 10, 'price': .50},
                          "CANDY": {'count': 10, 'price': .65}};
    }

    render(coin){
        if(this.areEqual(coin, COINS['NICKEL'])){
            return 5;
        }
        else if(this.areEqual(coin, COINS['DIME'])){
                return 10;
        }
        else if(this.areEqual(coin, COINS['QUARTER'])){
            return 25;
        }

    }

    insertCoin(coin){
        let value = this.render(coin);
        this.changeIn = this.changeIn + value;
    }

    getChangeIn(){
        return this.changeIn;
    }

    areEqual(coin1, coin2){
       if(coin1['Weight'] === coin2['Weight'] && coin1['Weight'] === coin2['Weight']){
           return true;
       }
       return false;
    }

    buyProduct(buttonPressed){
        if(buttonPressed === 'a' && this.enoughMoney(this.inventory['COLA']['price'])){
            this.inventory['COLA']['count'] = this.inventory['COLA']['count'] -1;
            return "COLA";
        }
        else if (buttonPressed === 'b' && this.enoughMoney(this.inventory['CHIP']['price'])){
            this.inventory['CHIP']['count'] = this.inventory['CHIP']['count'] -1;
            return "CHIP";
        }
        else if (buttonPressed === 'c' && this.enoughMoney(this.inventory['CANDY']['price'])) {
            this.inventory['CANDY']['count'] = this.inventory['CANDY']['count'] -1;
            return "CANDY";
        }

        if(buttonPressed === 'a' && this.changeIn > 1){
            return "EXACT CHANGE ONLY";
        }
        return "PLEASE INSERT MORE CHANGE";
    }

    enoughMoney(productPrice){
        if(productPrice === 1 && this.changeIn > productPrice){
            return false;
        }

        return this.changeIn >= productPrice;
    }

    colaCount(){
        return this.inventory['COLA']['count'];
    }

    chipCount(){
        return this.inventory['CHIP']['count'];
    }

    candyCount(){
        return this.inventory['CANDY']['count'];
    }





}